test/e2e/config/e2e_conf.yaml

```diff
@@ -70,21 +70,41 @@ providers:
       targetName: "cluster-template-centos.yaml"
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION}"
-  UPGRADED_K8S_VERSION: "${UPGRADED_K8S_VERSION}"
   CNI: "/tmp/calico.yaml"
+  KUBERNETES_VERSION: "v1.23.2"
+  CONTROL_PLANE_MACHINE_COUNT: 3
+  WORKER_MACHINE_COUNT: 1
   APIVersion: "infrastructure.cluster.x-k8s.io/${CAPM3_VERSION}"
   NAMEPREFIX: "${NAMEPREFIX:-baremetal-operator}"
   IRONIC_NAMESPACE: "${IRONIC_NAMESPACE:-baremetal-operator-system}"
   CONTAINER_REGISTRY: "${CONTAINER_REGISTRY:-quay.io}"
   IRONIC_IMAGE_TAG: "${IRONIC_IMAGE_TAG:-main}"
   MARIADB_IMAGE_TAG: "${MARIADB_IMAGE_TAG:-main}"
   UPGRADED_BMO_IMAGE_TAG: "${UPGRADED_BMO_IMAGE_TAG:-main}"
```

Comment:
Can we use here only explicit values check this comment https://github.com/metal3-io/cluster-api-provider-metal3/pull/550/files#r819782996

Reply:
I'm not sure exactly how this works in all situations. I guess we could just put `main` here and then the e2e framework would still take the value from the environment variable if it is set? If that's the case then yes we should just put `main` here.

Reply:
I tried to limit the changes to only those affecting the dependency on devenv so I prefer we add these improvements separately